### ISSUE TITLE
[A11y] General - login modal

### ISSUE THREAD

**Issue by jechols** [OregonDigital/OD2#2035]
### Operating System

Windows

### Browser

Edge

### Affected page

Homepage

### Admin user?

No

### Issue Description

Odd regression here as of the 9/3 deploy (I think -- though maybe this has been here the whole time and I never noticed): you can use the keyboard to get to the various radio buttons and the login button once you select the radio button you want **except for unaffiliated people**. When using the arrow keys to select "Other Affiliation", tabbing brings you to the "Close" button, not the "CONTINUE TO LOG IN" button.

Even more odd is that shift-tabbing works to get to that button. It actually properly wraps back to the beginning of the dialog. But shift-tab doesn't work for the rest of the modal.

### Suggested Fix

Make sure all buttons are "inside" the dialog no matter which radio button is selected. I suspect there are still some logic bugs with the dialog's modal behaviors in how it implements tab and shift-tab to "wrap" focus.

### Can anybody QA or does this need an expert?

Anybody

### QA Steps

- Tab to the "log in" button and select it (enter or space)
- Tab to the radio buttons
- Use left and right arrows to select a radio button
- Make sure that from each radio button, you can tab to the appropriate "CONTINUE TO..." button
- Make sure shift-tab and tab always "wrap" inside the dialog. If tabbing would take you past the last element, it should return to the first, and the opposite should be true of shift-tabbing.

**Comment by thornka** [comment]
Duplicate of #1790

**Comment by azle** [comment]
Updated  12/26/2024

Chrome on MAC

As a user, when I tab to the radio buttons for OSU, UO and Other Affiliation it forces the user to tab to close before moving on to the next radio button and limits the user from being able to use the left and right arrows. You still have to tab close and you can only use 1 left or right arrow to move before it goes to close again (similiar to the tab issue).

For example: tab OSU tab Close, tab UO tab Close tab Other Affiliation tab Close

Standard practice would be that the user:

Can tab to close > tab through the selection of radio buttons OSU, UO and Other Affilation > then tab Continue to Login and Contact Us.
